### PR TITLE
feat: filter out tranches the user has not invested in

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This PR will:

- on the show page: filter out the tranches that the user has not invested in

## Checklist

- [ ] I used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
- [ ] Unit tests have been created if a new feature was added.
- [ ] Translations have been added and extracted if a new feature was added.

### CODE REVIEW

File: src/components/DealInvestments.tsx [hunk 1]
```diff
@@ -21,13 +47,14 @@ export const DealInvestments: FunctionComponent<DealInvestmentsProps> = ({ class
 					description: "Deal investments: title",
```
Comment (MaximPiessen) [inline src/components/DealInvestments.tsx]: can we also remove the 'your investments' title when you didn't invest in any tranches (when the deal is in progress/ended)?

File: src/components/DealInvestments.tsx [hunk 2]
```diff
@@ -21,13 +51,14 @@ export const DealInvestments: FunctionComponent<DealInvestmentsProps> = ({ class
...
 				})}
 			</div>
-			{deal.tranches?.tranches
+			{tranches
 				// Remove Senior tranche from array
 				.filter((t) => t.index > 1)
```
Comment (MaximPiessen) [inline src/components/DealInvestments.tsx]: != 1

super-senior tranche should appear if its size != 0

File: src/components/DealInvestments.tsx [hunk 3]
```diff
@@ -12,6 +16,32 @@ interface DealInvestmentsProps {
...
+	const [tranches, setTranches] = useState<Tranche[]>([]);
+	const [userTrancheBalances, setUserTrancheBalances] = useState<{ index: TokenAmount }>();
+	const { publicKey } = useWallet();
+
+	useEffect(() => {
+		const getInvestedTranches = async () => {
+			const balances = {} as { index: TokenAmount };
+			const tranches = await asyncFilter(deal.tranches.tranches, async (tranche) => {
+				try {
+					const userTrancheBalance = await tranche.userTrancheBalance(publicKey);
```
Comment (RuvendeGroote) [inline src/components/DealInvestments.tsx:29]: if we are doing it for every tranche and it's still a separate call for every tranche, I prefer having this logic just be part of the TrancheInvestment component

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

File: src/components/DealInvestments.tsx (hunk 3 above)
Comment (BartDeCaluwe) [reply src/components/DealInvestments.tsx:28]: So just return null then in the TrancheInvestment component if there is no userTrancheBalance?

Comment (BartDeCaluwe) [reply src/components/DealInvestments.tsx:29]: So just return null then in the TrancheInvestment component if there is no userTrancheBalance?

**Review by RuvendeGroote — DISMISSED** [review]

**Review by MaximPiessen — APPROVED** [review]

**Pull request merged**